hw2/01-coins.js

```diff
@@ -1,7 +1,52 @@
 /** Exercise 01 - Coins **/
 
 const calculateChange = (input) => {
-  // Add your code here
+  if (input > 10) return "Error: the number is too large"
+
+  let amountInCents = input * 100; // 4.62 -> 462
+
+  let dollars = Math.floor(amountInCents/100); // 4
+  amountInCents = amountInCents - (dollars*100); // 62
+
+  let quarters = Math.floor(amountInCents/25); //2
+  amountInCents = amountInCents - (quarters * 25); // 12
+
+  let dimes = Math.floor(amountInCents/10); // 1
+  amountInCents = amountInCents - (dimes * 10); // 2
+
+  let nickels = Math.floor(amountInCents/5); // 0
+  amountInCents = amountInCents - (nickels * 5); // 2
+
+  let pennies = amountInCents;
+  
+  let dollarString = "";
+  let quarterString = "";
+  let dimeString = "";
+  let nickelString = "";
+  let penniesString = "";
+
+  dollarString = dollars == 0 ? "" 
+                : dollars == 1 ? dollars.toString() + " dollar, "
+                : dollars.toString() + " dollars, ";
+
```

Comment:
I love how efficient this code is! I think your method of dividing the amount by the value and then doing a floor is so clever. I wrote mine almost like how you would count them in real life. I think yours is very elegant. The only thing I could suggest is encapsulating some of this into functions.

Code produces expected output!